libcrux-psq/src/handshake/ciphersuite.rs

```diff
@@ -0,0 +1,747 @@
+            #[cfg(not(feature = "classic-mceliece"))]
+            InitiatorCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(_) => {
+                panic!("unsupported ciphersuite")
+            }
+        }
+    }
+
+    fn is_pq(&self) -> bool {
+        match self {
+            InitiatorCiphersuite::X25519_ChaChaPoly_HkdfSha256(_) => false,
+            InitiatorCiphersuite::X25519_MlKem768_ChaChaPoly_HkdfSha256(_) => true,
+            #[cfg(feature = "classic-mceliece")]
+            InitiatorCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(_) => true,
+            #[cfg(not(feature = "classic-mceliece"))]
+            InitiatorCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(_) => {
+                panic!("unsupported ciphersuite")
+            }
+        }
+    }
+
+    fn pq_encapsulate(
+        &self,
+        rng: &mut impl CryptoRng,
+    ) -> Result<
+        (
+            Option<<Self as CiphersuiteBase>::Ciphertext>,
+            Option<<Self as CiphersuiteBase>::SharedSecret>,
+        ),
+        HandshakeError,
+    > {
+        match self {
+            InitiatorCiphersuite::X25519_ChaChaPoly_HkdfSha256(_) => Ok((None, None)),
+            InitiatorCiphersuite::X25519_MlKem768_ChaChaPoly_HkdfSha256(
+                initiator_x25519_mlkem768_chacha_poly_hkdf_sha256,
+            ) => {
+                let mut rand = [0u8; libcrux_ml_kem::ENCAPS_SEED_SIZE];
+                rng.fill_bytes(&mut rand);
+                let (ct, ss) = libcrux_ml_kem::mlkem768::encapsulate(
+                    initiator_x25519_mlkem768_chacha_poly_hkdf_sha256.peer_longterm_mlkem_pk,
+                    rand,
+                );
+
+                Ok((
+                    Some(DynamicCiphertext::MlKem(ct)),
+                    Some(DynamicSharedSecret::MlKem(ss)),
+                ))
+            }
+            #[cfg(feature = "classic-mceliece")]
+            InitiatorCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(
+                initiator_x25519_classic_mc_eliece_chacha_poly_hkdf_sha256,
+            ) => {
+                use crate::classic_mceliece::ClassicMcEliece;
+                use libcrux_traits::kem::KEM;
+
+                let (ss, ct) = <ClassicMcEliece as KEM>::encapsulate(
+                    &initiator_x25519_classic_mc_eliece_chacha_poly_hkdf_sha256
+                        .peer_longterm_cmc_pk,
+                    rng,
+                )
+                .map_err(|_| HandshakeError::CryptoError)?;
+
+                Ok((
+                    Some(DynamicCiphertext::CMC(ct)),
+                    Some(DynamicSharedSecret::CMC(ss)),
+                ))
+            }
+
+            #[cfg(not(feature = "classic-mceliece"))]
+            InitiatorCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(_) => {
+                panic!("unsupported ciphersuite")
+            }
+        }
+    }
+}
+
+impl<'a> CiphersuiteBuilder<'a> {
+    pub fn new() -> Self {
```

Comment:
For builders you usually add a `builder()` function to the resulting object as well to make it more seamless. Maybe that's difficult here because of the different results. Also not too important. Just a note.